src/mixins/createCachedRenderMixin.ts

```diff
@@ -121,9 +126,15 @@ const createCachedRenderMixin: CachedRenderFactory = createStateful
 					props[key] = cachedRender.listeners[key];
 				}
 				const classes: { [index: string]: boolean; } = {};
+				const widgetClasses: string[] = widgetClassesMap.get(cachedRender) || [];
+
+				widgetClasses.forEach((c) => classes[c] = false);
+
 				if (cachedRender.classes) {
 					cachedRender.classes.forEach((c) => classes[c] = true);
 				}
+
+				widgetClassesMap.set(cachedRender, cachedRender.classes || []);
```

Comment:
Ah, I didn't catch that, but don't you want the list to continue to persist from render to render... You never want it to drop out, so shouldn't you be re-using the array you got from above and just pushing any uniques onto it?